server/src/executor/mod.rs

```diff
@@ -130,10 +141,45 @@ impl Executor {
         }
     }
 
+    fn handle_respawn(
+        &self,
+        client_id: u32,
+        game_state: &mut GameState,
+        physics_state: &mut PhysicsState,
+        game_events: &mut dyn GameEventCollector,
+    ) {
+        // if dead player is not on the respawn map, insert it into it
+        if let Some(player) = game_state.players.get(&client_id) {
+            if !player.on_cooldown.contains_key(&Command::Respawn) {
+                game_state.insert_cooldown(client_id, Command::Respawn, 3);
+            }
+        }
+        // Update all cooldowns in the game state
+        game_state.update_cooldowns();
```

Comment:
why update cooldowns inside `handle_respawn`? Is the cooldown logic coupled with the respawn logic?

Reply:
Yes, previously we didn't have it here. But then, we found that the logic of respawn is kinda different from other normal cd, like jump, attack, etc. Thus, we change the logic here and refactor it

Reply:
oh so is the cooldown for respawn a separate from other ones, so we can't just update the cooldown after each tick?

Reply:
yeah exactly. Since we want it to wait for respawning everytime players's dead. But if using the same logic, the firsttime death will trigger respwn immediately